src/audio/dai.c

```diff
@@ -889,8 +889,8 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 
 	/* cannot configure DAI while active */
 	if (dev->state == COMP_STATE_ACTIVE) {
-		comp_err(dev, "dai_config(): Component is in active state.");
-		return -EINVAL;
+		comp_info(dev, "dai_config(): Component is in active state. Ignore config");
+		return 0;
```

Comment:
This should not be changed, it should not be configured the 2nd times for the same dev/direction.

Reply:
@ranj063 do we support dynamic pipeline loading with main features? Or this is triggered by fuzzer? 
Dynamic pipeline loading would an awesome feature in general.
In the general case of dynamic pipeline loading i don't think this would be an issue because before loading a new pipeline will basically unload the currrent pipeline and the state of the device will no longer be inited.

Now for your particular case, where you have separate loads for playback and capture I think we need a better solution, maybe set a state per direction.

Reply:
@dbaluta yes I'm in the process of implementing it. https://github.com/thesofproject/linux/pull/2371

Reply:
@keyonjie agree it should not but when u configure the dai, all the info you've got in the config is type and dai index, which are the same for both DAIs. So you end up configuring them twice today, once each per direction

Reply:
> Now for your particular case, where you have separate loads for playback and capture I think we need a better solution, maybe set a state per direction.

But SSP config is the same irrespective of direction. State per direction doesnt seem correct either. On the other hand DAIs are per direction but today there is no way for you to look at the DAI dev and figure out whether it is for playback/capture

Reply:
@keyonjie any comments - we can discuss on call if needed ?

Reply:
@ranj063 SSP is a special case for the dai config is shared by playback & capture. We should not set dai config twice or more for other dai type, so we may need to check dai type here for SSP only

Reply:
@RanderWang you have a valid point but it is only the SSP that actually has the same DAI index for the 2 directions. For the others, we have separate indices and hence separate DAIs, so theoretically we'd never end up in a situation like this. But I will add the check for type to make it clearer.

Reply:
@ranj063 Should the config be just ignored or verified with the previously applied one for consistency? Does topology creation process guarantee that settings are not inconsistent?

Reply:
@mmaka1 as far as topology is concerned, there is only one config for both directions. The only reason we need the change in the firmware is because the DAI components are separate for the 2 directions. But when creating and configuring the first DAI, the common config will have already been applied so we can safely ignore the second request fo configure the SSP.